dataprep/clean/clean_duplication_utils.py

```diff
@@ -28,6 +29,8 @@
     }
 """
 
+DF_FILE = path.join(path.split(path.abspath(__file__))[0], "clean_duplication_output.pkl.gz")
```

Comment:
1. Do we know if gzip is faster? If not it would better to test if gzip do give us a speed boost
2. use `tempfile.TemporaryDirectory` so that we don't corrupt the user's work directory. Using tempfile will also be faster because on Linux temp folder is usually in memory.

Reply:
Thanks for the comments @dovahcrow, I made the requested changes. 
1. Using gzip is slower, I was trying to save space but it probably doesn't matter since it's only temporary
2. Good idea, I couldn't use the `TemporaryDirectory` class as a context manager since I don't think there's a way to pass the object into the code that runs in the IPython kernel. I think I was able to achieve the same thing by using `tempfile.mkdtemp()` and manually deleting the directory afterwards.

Reply:
@ryanwdale great! I think the `mkdtemp` solution also works.